src/sth_server.js

```diff
@@ -516,36 +516,42 @@
-                  if (isNaN(attributes[j].value)) {
-                    // The attribute value is not a number and consequently not able to be aggregated.
-                    sthLogger.fatal('Attribute value not aggregatable', {
-                      operationType: sthConfig.OPERATION_TYPE.SERVER_LOG
-                    });
-                    continue;
+            if (totalAttributes > 0) {
+              for (var i = 0; i < contextResponses.length; i++) {
+                if (contextResponses[i].contextElement &&
+                  contextResponses[i].contextElement.attributes &&
+                  Array.isArray(contextResponses[i].contextElement.attributes)) {
+                  contextElement = contextResponses[i].contextElement;
+                  attributes = contextElement.attributes;
+                  for (var j = 0; j < attributes.length; j++) {
+                    if (isNaN(attributes[j].value)) {
+                      // The attribute value is not a number and consequently not able to be aggregated.
+                      sthLogger.fatal('Attribute value not aggregatable', {
+                        operationType: sthConfig.OPERATION_TYPE.SERVER_LOG
+                      });
+                      continue;
+                    }
+
+                    attribute = attributes[j];
+
+                    // Compose the collection name for the required data
+                    var databaseName = sthDatabase.getDatabase(
+                      request.headers['fiware-service']);
+
+                    var servicePath = request.headers['fiware-servicepath'];
+
+                    // Store the raw data into the database
+                    storeRawData(databaseName, servicePath, contextElement, attribute, recvTime, counterObj, totalTasks, request, reply);
+
+                    // Store the aggregated data into the database
+                    storeAggregatedData(databaseName, servicePath, contextElement, attribute, recvTime, counterObj, totalTasks, request, reply);
                   }
-
-                  attribute = attributes[j];
-
-                  // Compose the collection name for the required data
-                  var databaseName = sthDatabase.getDatabase(
-                    request.headers['fiware-service']);
-
-                  var servicePath = request.headers['fiware-servicepath'];
-
-                  // Store the raw data into the database
-                  storeRawData(databaseName, servicePath, contextElement, attribute, recvTime, counterObj, totalTasks, request, reply);
-
-                  // Store the aggregated data into the database
-                  storeAggregatedData(databaseName, servicePath, contextElement, attribute, recvTime, counterObj, totalTasks, request, reply);
                 }
               }
+            } else {
+              var error = boom.badRequest('At least one attribute with an aggregatable value should be included in the notification');
```

Comment:
Well, it depends if we want the requester (in this case the Context Broker) to know that it is doing something wrong (basically, the user may have created a subscription to a not aggregatable attribute)... :D The only way "to know" it would be to make a query and get no results since nothing has been stored in the STH databases (I think the last thing the user is going to think about is that the attribute had non-aggregatable values :) ) What do you think? Do you still think we should be just replying with a 200 - OK? Or maybe any other HTTP status code? :)